clients/fides-js/src/lib/gpp/us-notices.ts

```diff
@@ -0,0 +1,92 @@
+/**
+ * Helper functions to set the GPP CMP API based on Fides values
+ */
+
+import { CmpApi } from "@iabgpp/cmpapi";
+
+import { FidesCookie } from "../cookie";
+import {
+  FIDES_REGION_TO_GPP_SECTION,
+  NOTICE_KEY_TO_FIDES_REGION_GPP_FIELDS,
+} from "./constants";
+import { PrivacyExperience } from "../consent-types";
+
+/**
+ * Sets the appropriate fields on a GPP CMP API model for whether notices were provided
+ *
+ * Returns GPP section IDs which were updated
+ */
+export const setGppNoticesProvidedFromExperience = ({
+  cmpApi,
+  experience,
+}: {
+  cmpApi: CmpApi;
+  experience: PrivacyExperience;
+}) => {
+  const sectionsChanged = new Set<{ name: string; id: number }>();
+  const { privacy_notices: notices, region } = experience;
+
+  Object.entries(NOTICE_KEY_TO_FIDES_REGION_GPP_FIELDS).forEach(
+    ([noticeKey, regionGppFields]) => {
+      const gppFields = regionGppFields[region];
+      if (gppFields) {
+        const { gpp_notice_fields: fields } = gppFields;
+        const experienceHasNotice = !!notices?.find(
+          (n) => n.notice_key === noticeKey
```

Comment:
I'm trying to get a sense for how GPP + TCF will interact. So if a user's experience / cookie has a consent val for a notice_key of "data_sales_and_sharing", then this consent val would apply for both GPP and TCF right?

Reply:
not quite— `data_sales_and_sharing` is a privacy notice, and TCF doesn't use privacy notices (it uses all those `tcf_*` objects instead). so for TCF, we don't get any privacy notices back, AND the region isn't one of the regions defined in that `constants.ts` file, and so `if (gppFields)` wouldn't trigger (there are no gpp fields for TCF) so I guess the flow in this particular bit of logic is...

1. Map through all privacy notices
2. Grab the GPP fields that apply for our region
3. In the case of TCF, no GPP fields apply in i.e. France (this is defined by `NOTICE_KEY_TO_FIDES_REGION_GPP_FIELDS`)
4. so nothing happens in this function

Overall, we can think of GPP as another way to represent TCF _and_ a whole bunch of other policies. They "interact" if TCF is being used (i.e. in an EU region), and in that case, our GPP code subscribes to TCF updates, and sets the TC string that the TCF API gives us. For US regions (non-TCF), each state has a different policy which is encoded more or less in `NOTICE_KEY_TO_FIDES_REGION_GPP_FIELDS`. there's no region (right now...) that uses both TCF and another policy, so in theory they are separate

Reply:
oh thanks for the clarification! Could you point me to the bit of code where `our GPP code subscribes to TCF updates, and sets the TC string that the TCF API gives us`?

Reply:
yes! I suppose "GPP code subscribes to TCF updates" isn't quite right—it really subscribes to Fides updates, but that includes the TC string. 

https://github.com/ethyca/fides/blob/9470384dd0c37a6eaec8532a3030cd515ce48a44/clients/fides-js/src/fides-ext-gpp.ts#L143-L150